### ISSUE TITLE
Video latency minimization

### ISSUE THREAD

**Issue by zarik5** [JackD83/ALVR#251]
Video latency minimization with a statistical model. The same model can be used in multiple ways to mitigate latency and jitter. It does not require time sync between server and client because it self-corrects.
* [ ] Create EventTiming
* [ ] Apply model for total pipeline latency reduction
* [ ] Apply model for black pull reduction and total latency estimation
* [ ] Apply model for headset jitter. Controller jitter cannot be reduced due to a limitation of the OpenVR driver API.

**Comment by fangguanya** [comment]
@zarik5 Hi, Could you please explain further about why was controller jitter cannot reduced for me? Thanks. I didn't know about the difference between HMD tracking with Controller tracking well.

**Comment by zarik5** [comment]
Maybe it's possible to have more control over the controller jitter if we disable the OpenVR controller prediction algorithm. I'll see what I can do when I get to that point.

**Comment by fangguanya** [comment]
> Maybe it's possible to have more control over the controller jitter if we disable the OpenVR controller prediction algorithm. I'll see what I can do when I get to that point.

Cann't we just use the same way that we treat with sync of HMD position?

**Comment by zarik5** [comment]
No, because OpenVR frame timing information contains info only for the display reprojection. If we let SteamVR do its controller prediction then we don't have correspondence between video and tracking frame anymore, thus we get judder. Anyway, I don't have everything fresh in mind, I'll see what I can do in the next major releases.

**Comment by fangguanya** [comment]
Great!
